elasticsearch/templates/deployment-es-client.yaml

```diff
@@ -0,0 +1,72 @@
+# Copyright 2017 The Openstack-Helm Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ .Values.labels.es_client.name }}
+  labels:
+    component: {{ .Values.labels.es_client.component }}
+    role: {{ .Values.labels.es_client.role }}
+spec:
+  replicas: {{ .Values.replicas.es_client }}
+  template:
+    metadata:
+      labels:
+        component: {{ .Values.labels.es_client.component }}
+        role: {{ .Values.labels.es_client.role }}
+    spec:
+      containers:
+      - name: {{ .Values.labels.es_client.name }}
+        securityContext:
+          privileged: false
+          capabilities:
+            add:
+              - IPC_LOCK
+              - SYS_RESOURCE
+        image: {{ .Values.images.elasticsearch }}
+        imagePullPolicy: {{ .Values.images.pullPolicy }}
+        env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: "CLUSTER_NAME"
+          value: {{ .Values.cluster_name }}
+        - name: NODE_MASTER
+          value: "false"
+        - name: NODE_DATA
+          value: "false"
+        - name: HTTP_ENABLE
+          value: "true"
+        - name: "ES_JAVA_OPTS"
+          value: {{ .Values.es_java_opts.es_client }}
+        ports:
+        - containerPort: {{ .Values.network.ports.http }}
+          name: http
+          protocol: TCP
+        - containerPort: {{ .Values.network.ports.elasticsearch }} 
+          name: transport
+          protocol: TCP
+        volumeMounts:
+        - name: storage
+          mountPath: /data
+      volumes:
```

Comment:
To use an empty directory for storage here, it should be

- name: storage
    emptyDir: {}